src/brsTypes/components/RoDeviceInfo.ts

```diff
@@ -0,0 +1,759 @@
+            return new BrsString("");
+        },
+    });
+
+    private getModelDetails = new Callable("getModelDetails", {
+        signature: {
+            args: [],
+            returns: ValueKind.Object,
+        },
+        impl: (_interpreter) => {
+            return new RoAssociativeArray([]);
+        },
+    });
+
+    private getFriendlyName = new Callable("getFriendlyName", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("");
+        },
+    });
+
+    private getOSVersion = new Callable("getOSVersion", {
+        signature: {
+            args: [],
+            returns: ValueKind.Object,
+        },
+        impl: (_interpreter) => {
+            return new RoAssociativeArray([]);
+        },
+    });
+
+    private getVersion = new Callable("getVersion", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("");
+        },
+    });
+
+    private getRIDA = new Callable("getRIDA", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("");
+        },
+    });
+
+    private isRIDADisabled = new Callable("isRIDADisabled", {
+        signature: {
+            args: [],
+            returns: ValueKind.Boolean,
+        },
+        impl: (_interpreter) => {
+            return BrsBoolean.True;
+        },
+    });
+
+    private getChannelClientId = new Callable("getChannelClientId", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("");
+        },
+    });
+
+    private getUserCountryCode = new Callable("getUserCountryCode", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            let countryCode = process.env.LOCALE;
+            return countryCode ? new BrsString(countryCode) : BrsInvalid.Instance;
+        },
+    });
+
+    private getRandomUUID = new Callable("getRandomUUID", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            let uuid = uuidv4();
+            return new BrsString(uuid);
+        },
+    });
+
+    private getTimeZone = new Callable("getTimeZone", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            let timeZone = process.env.TZ;
+            return timeZone ? new BrsString(timeZone) : BrsInvalid.Instance;
+        },
+    });
+
+    private hasFeature = new Callable("hasFeature", {
+        signature: {
+            args: [new StdlibArgument("feature", ValueKind.String)],
```

Comment:
nit, please add a whitespace between brackets

Reply:
@alimnios72  - It looks like the code gets auto-prettified when I make a commit so it undos the whitespace changes I make to the brackets. :(

Reply:
oh ok, we'll have to leave it that way :)